static/slider.js

```diff
@@ -90,56 +111,99 @@
                         allStars[construct] += filtered.length;
                     }
                 } else {
                     observationsFiltered[construct][course] = {};
+                    horizontalCourse[construct][course] = {};
 
                     for (var sublevel in observations[construct][course]) {
                         const filtered = observations[construct][course][sublevel]
                             .filter(date => date <= timestamp);
 
                         observationsFiltered[construct][course][sublevel] = filtered;
+                        horizontalCourse[construct][course][sublevel] = filtered;
                         allStars[construct] += observationsFiltered[construct][course][sublevel].length;
                     }
                 }
             }
         }
 
-        // Swap verticalStarChart content if any of construct was vertically merged.
+        // Swap verticalStarChart and horizontalStarChart content if any of construct was merged.
         swapVerticalStarChart(courseObservations, allStars);
+        swapHorizontalStarChart(horizontalCourse, allStars);
 
         // Change elements in table.
-        changeElements(observationsFiltered, allStars);
+        changeElements(observationsFiltered, allStars, joinMergedSublevels(observationsFiltered));
     }
 
     /**
      * Changes colored cells and those with observations count after filtering observations by date.
      * @param {Object} observationsFiltered - Observations filtered by date.
      * @param {Object} allStars - Observations count per construct.
      */
-    function changeElements(observationsFiltered, allStars) {
+    function changeElements(observationsFiltered, allStars, levels) {
         for (var construct in observationsFiltered) {
             for (var course in observationsFiltered[construct]) {
                 for (var sublevel in observationsFiltered[construct][course]) {
-                    const size = observationsFiltered[construct][course][sublevel].length;
-                    const color = calculateNewColor(size, allStars[construct]);
-
-                    // If course is 0 then it means construct was merged.
-                    if (parseInt(course)) {
-                        changeElems(size, color, `.stars-${course}-${sublevel}`,
-                            `.heat-${course}-${sublevel}`, construct);
-                    } else {
-                        changeElems(size, color, `.stars-merged-${sublevel}`,
-                            `.heat-merged-${sublevel}`, construct);
-                    }
+                    changeSublevel(
+                        observationsFiltered,
+                        course,
+                        sublevel,
+                        construct,
+                        allStars,
+                        levels
+                    );
                 }
             }
         }
     }
 
+    /**
+     * Change elements for single sublevel or level.
```

Comment:
What do you change here? You pass here a lot o arguments. Please describe it a little more verbose.